### PR TITLE
fix: hash-in-strings, triple-quoted strings, -bor in args, generic arity types

### PR DESCRIPTION
**Pull request by tinovyatkin**
## Summary

- **`#` in expandable strings**: `#` after variable/subexpression interpolation (e.g. `"$var # text"`) was consumed by the `comment` extra, causing catastrophic parse failures in PSService.ps1 and ShadowCopy.ps1. Fixed via external scanner token that bypasses extras inside string content.
- **Triple-quoted strings**: `"""$s2"""` failed because `""` matched as empty string first. Reordered choices in `expandable_string_literal` and extended `_expandable_string_leading_text` to match `""`.
- **`-bor`/`-band`/`-bxor` in argument lists**: `bitwise_argument_expression` used `-and`/`-or`/`-xor` (logical operators) instead of the bitwise variants. Copy-paste bug from the non-argument expression rule.
- **Double-backtick generic arity**: `[System.Collections.Generic.List``1]` failed because `type_identifier` only allowed single backtick. Extended to accept 1-2 backticks.
- Removed now-unnecessary `expandable_string_literal` conflict declaration.
- Added 4 fixture files from [JFLarvoire/SysToolsLib](https://github.com/JFLarvoire/SysToolsLib/tree/master/PowerShell) — all parse at 0 errors.

### Before → After (SysToolsLib files)

| File | Before | After |
|------|--------|-------|
| EchoArgs.ps1 (8.8 KB) | 2 errors | **0** |
| PSService.ps1 (63 KB) | 3 errors (whole-file) | **0** |
| ShadowCopy.ps1 (103 KB) | 5 errors (whole-file) | **0** |
| Window.ps1 (45 KB) | 3 errors | **0** |

## Test plan

- [x] All 217 corpus tests pass (5 new tests added, 4 error-recovery tests updated)
- [x] All 9 fixture files parse at 0 errors
- [x] Go tests pass
- [x] `LANGUAGE_VERSION` remains 14 (ABI compat)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **Bug Fixes**
  * Fixed parsing of bitwise operators to use correct `-band/-bor/-bxor` syntax

* **New Features**
  * Added support for double-backtick qualified generic type identifiers
  * Improved expandable string parsing to correctly handle edge cases with interpolations

* **Tests**
  * Expanded test coverage for bitwise operations and generic type syntax
  * Added test cases for complex expandable string patterns

### CODE REVIEW

File: src/scanner.c [hunk 1]
```diff
@@ -139,12 +140,34 @@ static bool scan_for_clause_break(TSLexer *lexer, const bool *valid_symbols)
...
+{
+    if (!valid_symbols[EXPANDABLE_STRING_IMMCONTENT]) return false;
+
+    bool advanced = false;
+    while (lexer->lookahead != 0 &&
+           lexer->lookahead != '$' &&
+           lexer->lookahead != '`' &&
+           lexer->lookahead != '"' &&
+           lexer->lookahead != '\r' &&
+           lexer->lookahead != '\n') {
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply src/scanner.c]: ✅ Addressed in 017b79f. Thanks @gemini-code-assist!
